modules/concurrency/include/hephaestus/concurrency/when_all_range.h

```diff
@@ -0,0 +1,168 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <memory>
+#include <optional>
+#include <ranges>
+#include <type_traits>
+#include <utility>
+
+#include <stdexec/__detail/__execution_fwd.hpp>
+#include <stdexec/__detail/__senders_core.hpp>
+#include <stdexec/execution.hpp>
+
+namespace heph::concurrency {
+template <typename Range>
+concept SenderRange = std::ranges::range<Range> && stdexec::sender<std::ranges::range_value_t<Range>>;
+
+namespace internal {
+
+struct WhenAllStopCallback {
+  stdexec::inplace_stop_source* stop_source;
+  void operator()() const {
+    stop_source->request_stop();
+  }
+};
+
+enum WhenAllRangeState : std::uint8_t { STARTED, ERROR, STOPPED };
+
+template <typename Receiver>
+struct WhenAllRangeStateT {
+  using ReceiverEnv = stdexec::env_of_t<Receiver>;
+  using StopToken = stdexec::stop_token_of_t<ReceiverEnv>;
+  using StopCallbackT = stdexec::stop_callback_for_t<StopToken, WhenAllStopCallback>;
+
+  void arrive() noexcept {
+    if (1 == count.fetch_sub(1, std::memory_order_acq_rel)) {
+      complete();
+    }
+  }
+
+  void complete() noexcept {
+    on_stop.reset();
+
+    switch (state.load(std::memory_order_relaxed)) {
+      case STARTED:
+        stdexec::set_value(std::move(receiver));
+        break;
+      case ERROR:
+        stdexec::set_error(std::move(receiver), std::move(error));
+        break;
+      case STOPPED:
+        stdexec::set_stopped(std::move(receiver));
+    }
+  }
+
+  [[nodiscard]] auto getEnv() const noexcept {
+    return stdexec::__env::__join(stdexec::prop{ stdexec::get_stop_token, stop_source.get_token() },
+                                  stdexec::get_env(receiver));
+  }
+
+  Receiver receiver;
+  std::atomic<std::size_t> count{ 0 };
+  stdexec::inplace_stop_source stop_source;
+  std::atomic<WhenAllRangeState> state{ STARTED };
+  std::exception_ptr error;
+  std::optional<StopCallbackT> on_stop;
+};
+
+template <typename OuterReceiver>
+struct InnerReceiver {
+  using receiver_concept = stdexec::receiver_t;
+
+  //  NOLINTBEGIN (readability-identifier-naming) - wrapping stdexec interface
```

Comment:
missing const doesnt get flagged
```suggestion
  //  NOLINTBEGIN(readability-identifier-naming) - wrapping stdexec interface
```